pyproject.toml

```diff
@@ -54,7 +54,12 @@ packages = [{include = "cai_causal_graph"}]
 
 [tool.poetry.dependencies]
 networkx = ">=3.0.0, <4.0.0"
-numpy = "^1.18.0"
+# Newest numpy is >= 3.9 so need to have different deps while we still support 3.8 but will drop 3.8 on next minor.
+numpy = [
+    {version = "^1.20.0", python = ">=3.12.0, <3.13.0"},  # 1.26 is the first version to support 3.12.
```

Comment:
Hey @andrew-causalens, if the first version to support 3.12 is 1.26, should this not say ^1.26.0?

Reply:
Poetry will resolve it to that. The reason stuff was failing before is because `numpy` didn't actually say older versions could not be used for 3.12 and it was trying to build older `numpy` from source and failing. However, this allows poetry to resolve a different version for 3.12. I didn't want to bump lower bound cause when we drop 3.8 support I can just return to `numpy = "^1.20.0"` and it will work. But right now there is no overlap of a single version that supports 3.8 - 3.12

Reply:
Changed it to allow one set of ranges for 3.9-3.12 and then another for 3.8. Great suggestion!